### ISSUE TITLE
Properly address property pages item ordering

### ISSUE THREAD

**Issue by MiYanni** [dotnet/project-system#7025]
Related: https://github.com/dotnet/project-system/issues/6989

Currently, CPS does not seem to maintain item ordering. This can be seen in the Target Framework dropdown. For now, we are putting in a hotfix to manually order these items. However, the proper fix has a couple options:

1. Investigate and fix CPS item ordering preservation
  a. For target framework, the items are declared here and this ordering should be preserved: https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedTargetFrameworks.props
2. Add functionality to the Rule definitions as to allow custom ordering.

These two options could both be completed, where the default is preserved ordering and it is optional to add your own ordering to the Rule definitions. When this PR is completed, please revert the hotfix here:

**Comment by tmeschter** [comment]
A third option would be to update Microsoft.NET.SupportedTargetFrameworks.props to include explicit ordering information separate from the order of items in the .props.

**Comment by tmeschter** [comment]
Fourth option: parse the Include for each item from that .props (e.g. `.NETCoreApp,Version=v2.1`) to group items by the framework family (e.g. .NETCoreApp) and then order by version number within that family.

**Comment by drewnoakes** [comment]
I would vote for adding specific `DisplayOrder` metadata to the items returned by the SDK.

**Comment by melytc** [comment]
> Fourth option: parse the Include for each item from that .props (e.g. `.NETCoreApp,Version=v2.1`) to group items by the framework family (e.g. .NETCoreApp) and then order by version number within that family.

To avoid parsing, this is "already done" in the SDK side: instead of capturing all of the values in SupportedTargetFramework item, we could store and use each framework's ItemGroup [Microsoft.NET.SupportedTargetFrameworks.props](https://github.com/sfoslund/sdk/blob/36876246b7dd181fe6d69d7fa68ba454c916aed4/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedTargetFrameworks.props).

**Comment by drewnoakes** [comment]
CPS now has a way for us to obtain the original item ordering, which may fix this.

We'll need code such as:

```c#
var after = projectChangeDescription.After.Items as IDataWithOriginalSource<KeyValuePair<string, IImmutableDictionary<string, string>>>;
```

This gives an `ICollection<KeyValuePair<string, IImmutableDictionary<string, string>>>` which has the original MSBuild order.

There are three places we could do this in the current code. Search for `https://github.com/dotnet/project-system/issues/7025` to find them.

**Comment by drewnoakes** [comment]
There's a cut of this in the `fix-7025-item-ordering` branch on my fork. It's blocked on needing the `IDataWithOriginalSource<>` interface on evaluation data, not just build data. I've asked CPS the feasibility of adding this support in `ProjectEvaluationSubscriptionService`.

**Comment by drewnoakes** [comment]
Evaluation data is handled differently to build data, and there's more re-use of the underlying immutable data structures over time. This means there's less likelihood we'd get ordered evaluation data using `IDataWithOriginalSource<>`. The approach should be fine for build data though.
